pinot-core/src/main/java/org/apache/pinot/core/segment/index/datasource/MutableDataSource.java

```diff
@@ -36,16 +36,26 @@
  */
 @SuppressWarnings("rawtypes")
 public class MutableDataSource extends BaseDataSource {
+  final boolean _fstIndexEnabled;
 
-  public MutableDataSource(FieldSpec fieldSpec, int numDocs, int numValues, int maxNumValuesPerMVEntry,
+  public MutableDataSource(FieldSpec fieldSpec, int numDocs, int numValues, int maxNumValuesPerMVEntry, boolean fstIndexEnabled,
       @Nullable PartitionFunction partitionFunction, @Nullable Set<Integer> partitions, @Nullable Comparable minValue,
       @Nullable Comparable maxValue, ForwardIndexReader forwardIndex, @Nullable Dictionary dictionary,
       @Nullable InvertedIndexReader invertedIndex, @Nullable InvertedIndexReader rangeIndex,
-      @Nullable TextIndexReader textIndex, @Nullable BloomFilterReader bloomFilter,
+      @Nullable TextIndexReader textIndex, @Nullable TextIndexReader fstIndex,
+      @Nullable BloomFilterReader bloomFilter,
       @Nullable NullValueVectorReader nullValueVector) {
     super(new MutableDataSourceMetadata(fieldSpec, numDocs, numValues, maxNumValuesPerMVEntry, partitionFunction,
-            partitions, minValue, maxValue), forwardIndex, dictionary, invertedIndex, rangeIndex, textIndex, bloomFilter,
-        nullValueVector);
+                    partitions, minValue, maxValue), forwardIndex, dictionary,
+            invertedIndex, rangeIndex, textIndex, fstIndex,
+            bloomFilter, nullValueVector);
+    _fstIndexEnabled = fstIndexEnabled;
+  }
+
+
+  // Returns whether the current field has
```

Comment:
Why this method specifically on mutable data source?

Reply:
This is to keep the results consistent between consuming & rolled out segments (more details in comments above)